WcaOnRails/app/controllers/api/v0/user_groups_controller.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class Api::V0::UserGroupsController < Api::V0::ApiController
+  before_action :current_user_is_authorized_for_action!, only: [:update]
+  private def current_user_is_authorized_for_action!
+    unless current_user.board_member? || current_user.admin?
```

Comment:
Use permissions method in the `user` class please

Reply:
(further down below you're even introducing a `can_access_board_panel` method which exactly fits this check)